### DISCUSSION TITLE
Better CPU performance when using Sandbox Mode: Chain Isolation vs Plugin Isolation (Beta 15)

### DISCUSSION THREAD

**Discussion by jaacMusic** [apohl79/audiogridder#1064] [Help] [answered]
Hi Andreas,

Thank you again for the great work!!
Here are a few findings on Beta 15:

I have noticed (at least with Beta 15) that when using multiple tracks (20+) with AG plugin hosting multiple plugins in a chain (4 or more), I see a difference from a 30-40% increase in CPU when switching from Sandbox Mode: Chain Isolation to Plugin Isolation. By the way, no sandboxing provides the best CPU saving.
Examples for a current project I'm running:
No Sandboxing: 51%
Chain Isolation: 59%
Plugin Isolation: 78%

I have a feeling this has to do with how many cores my AG server has or how the software handles hyperthreading.

Questions: 1) when using plugin isolation, is audio routed back to the DAW and passed on to the next plugin in the chain, or is the audio routed internally within the AG server and only routed back to the DAW after the last plugin in the chain? 2) If audio is routed back to the DAW for each plugin in the chain, wouldn't this introduce more round-trip latency?

Another thing, enabling "Use fixed size outbound buffers" still introduces audio distortion on my Cakewalk (Windows) system.

Thanks!

**Comment by apohl79** [comment, marked as answer]
Plugin isolation has more overhead than chain isolation due to the audio routing. The numbers you see is what I see too.

To your questions: the audio data is not routed back to the DAW for each plugin. It all happens within the server. But as there is a separate sandbox process for each plugin, the audio data has to be transmitted between all processes while processing the complete chain of plugins. This causes more I/O overhead and a bit higher response time.

> Another thing, enabling "Use fixed size outbound buffers" still introduces audio distortion on my Cakewalk (Windows) system.

ok, thats interesting. thx for letting me know.
